src/Operators/remapping.jl

```diff
@@ -138,110 +179,82 @@ function overlap_weights(
     return J_ov
 end
 
-function overlap(
-    target::T,
-    source::S,
-) where {
-    T <: SpectralElementSpace1D{<:IntervalTopology},
-    S <: SpectralElementSpace1D{<:IntervalTopology},
-}
-    QS_t = Spaces.quadrature_style(target)
-    QS_s = Spaces.quadrature_style(source)
-    Nq_t = Quadratures.degrees_of_freedom(QS_t)
-    Nq_s = Quadratures.degrees_of_freedom(QS_s)
-    if Nq_t >= Nq_s
```

Comment:
Moved this logic to within `x_overlap` and `y_overlap` - feels more transparent than having it here and then doing a matrix transposition. We may want to add a method that constructs the "inverse" map (target -> source) in practice, in which case transposing (and re-weighting) an existing (source -> target) remapping operator would be more efficient than building a new operator from scratch).

Reply:
Yes, sounds like a good idea to have a map inversion without having to build a new one from scratch

Reply:
Awesome. Can do that in a future PR.